src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+
+class RegistrationServiceImplTest {
+
+    private static RegistrationService registrationService;
+    private static List<User> validUsers;
+    private static List<User> unValidUsers;
+    private static List<User> nullFieldUsers;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+
+        validUsers = new ArrayList<>();
+        validUsers.add(new User(24L, "Sofia", "sofasofa", 25));
+        validUsers.add(new User(735L, "Bogdan", "bogdanchik", 19));
+        validUsers.add(new User(65449L, "Oleg", "olejik", 27));
+        validUsers.add(new User(1649L, "Ksenia", "kseniyaa", 26));
+        validUsers.add(new User(975L, "Dima", "demon228", 27));
+
+        unValidUsers = new ArrayList<>();
+        unValidUsers.add(new User(37L, "Valid", "validpassword", 25));
+        unValidUsers.add(new User(7305L, "Valid", "bogdanchik", 19));
+        unValidUsers.add(new User(679L, "Oleg", "olejik", 17));
+        unValidUsers.add(new User(169L, "Ksenia", "ksyu", 26));
+        unValidUsers.add(new User(856L, "", "demon228", 27));
+        unValidUsers.add(new User(-234L, "Peter", "peterparker", 25));
+
+        nullFieldUsers = new ArrayList<>();
+        nullFieldUsers.add(new User(null, "Bony", "bonybony", 27));
+        nullFieldUsers.add(new User(62345L, null, "davidbekham", 23));
+        nullFieldUsers.add(new User(73L,"Anna", null, 30));
+        nullFieldUsers.add(new User(623L, "Greg", "greg342", null));
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        User bob = new User(51323L, "Bob", "bobobob", -19);
+        assertThrows(RuntimeException.class, () -> registrationService.register(bob));
+    }
+
+    @Test
+    void register_nullFields_NotOk() {
+        for (User user : nullFieldUsers) {
+            assertThrows(RuntimeException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    void register_unValidUsers_NotOk() {
+        assertEquals(unValidUsers.get(0), registrationService.register(unValidUsers.get(0)));
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(1));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(2));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(3));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(4));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(5));
+        });
+    }
+
+    @Test
+    void register_fiveValidUsers_Ok() {
+        for (User valid : validUsers) {
+            assertEquals(valid, registrationService.register(valid));
+        }
+    }
+
+    @Test
+    void register_userIsAlreadyExist_NotOk() {
```

Comment:
and here you nead to add one user and check posibility to add user with the same login